### ISSUE TITLE
Improvements in train_test_split function

### ISSUE THREAD

**Issue by dshahid380** [aimacode/aima-python#1066]
### Issue Type
- Existing code improvement
### Proposed Changes
I would like to improve `train_test_split(dataset, start, end)` function existed in `learning.py` by adding training and testing set in percentage.
### Description
When i am working in `learning.py`, i found that the function `train_test_split` has three arguments-
- dataset
- start
- end 
<br>

Here `dataset.examples[start:end]` is used for testing and rest for training. I want to reframe the function in following way- <br>
`train_test_split(dataset, split = testing_split)`
#### Input
- dataset : Our dataset.
- split : Percentage by which it will be splitted in testing set.
#### Return
 - train : (100 - testing_split) * 100 % of dataset.examples.
 - val : testing_split * 100 % of dataset.examples. 
<br>
@MrDupin Can i start working on this?

**Comment by codefeeder** [comment]
### According to me, the splitting function should be modified for two kinds of data:
#### Time-Stamped data:
- The data provided should be sequentially divided into test and training set because then the model will essentially mimic the actual practice of using real-time data to predict future results.
- The split would be made according to the percentage passed as an attribute
### Data without a time-stamp
- The data should be picked up randomly and not in the original order provided to generate randomness in the train, validation and test set
- This would help in tackling overfitting and help in generalization

**Comment by dshahid380** [comment]
Hello @codefeeder, thanks for you kind suggestion. I think there is no time-stamped data for this repository but i will surely add an extra argument `shuffle=True` to select randomly training and testing data as you suggested.

**Comment by MrDupin** [comment]
Hello @dshahid380. I like this idea! I think it would be best if we had both options. Set start/end/percentage as optional arguments, and then choose which one to use depending on the passed arguments.

**Comment by codefeeder** [comment]
Hey @dshahid380, @MrDupin. May I contribute to this issue?

**Comment by dshahid380** [comment]
Hello @codefeeder, i am working on it.

**Comment by dshahid380** [comment]
@MrDupin Thank you for reviewing my issue. I have created a PR and did the same as you have suggested.
